src/instructlab/sdg/utils/chunkers.py

```diff
@@ -0,0 +1,563 @@
+        """Semantically chunk PDF documents.
+
+        Returns:
+            List: a list of chunks from the documents
+        """
+        if self.document_paths == []:
+            return []
+
+        model_artifacts_path = DocumentConverter.download_models_hf()
+        converter = DocumentConverter(artifacts_path=model_artifacts_path)
+        inputs = DocumentConversionInput.from_paths(self.filepaths)
+        parsed_documents = converter.convert(inputs)
+
+        docling_artifacts_path = self.export_documents(parsed_documents)
+
+        docling_json_paths = list(docling_artifacts_path.glob("*.json"))
+        chunks = []
+        for json_fp in docling_json_paths:
+            chunks.extend(self._process_parsed_docling_json(json_fp))
+
+        return chunks
+
+    def _path_validator(self, path) -> Path:
+        """
+        Validate the path and return a Path object.
+        Args:
+            path (str): Path to be validated.
+        Returns:
+            Path: Path object.
+        """
+        if isinstance(path, str):
+            path = Path(path)
+            if not path.exists():
+                raise FileNotFoundError(f"{path} does not exist.")
+        return path
+
+    def _load_qna_yaml(self, qna_yaml_path: Path) -> dict:
+        """
+        Load the qna YAML file.
+        Args:
+            qna_yaml_path (Path): Path to the knowledge qna YAML file.
+        Returns:
+            dict: Dictionary corresponding to knowledge qna YAML file.
+        """
+        if qna_yaml_path:
+            with open(qna_yaml_path, "r", encoding="utf-8") as f:
+                return yaml.safe_load(f)
+        return {}
+
+    def _process_parsed_docling_json(self, json_fp: Path) -> Dataset:
+        """
+        Process the parsed docling json file and return a dataset.
+        Args:
+            json_fp (Path): Path to the parsed docling json file.
+        Returns:
+            List: a list of chunks built from the provided json file
+        """
+        logger.info(f"Processing parsed docling json file: {json_fp}")
+        with open(json_fp, "r", encoding="utf-8") as f:
+            data = json.load(f)
+
+        file_name = json_fp.stem
+        chunks = self.build_chunks_from_docling_json(
+            data,
+            max_token_per_chunk=500,
+            tokenizer=self.tokenizer,
+        )
+        return self.fuse_texts(chunks, 200)
+
+    def fuse_texts(self, text_list, short_length_threshold=100):
+        """
+        Fuse short texts with preceding longer texts if their word count is below the threshold.
+        Args:
+            text_list (list): List of text chunks to process.
+            short_length_threshold (int): The word count threshold for determining short texts.
+        Returns:
+            list: List of fused texts.
+        """
+        fused_texts = []
+        previous_long_text = ""
+
+        for text in text_list:
+            word_count = len(text.split())
```

Comment:
Why is this using `len(text.split())` instead of just calling the tokenizer?  It seems odd to me to have a tokenizer for counting tokens during the initial chunking but then count words for the fusing.  Probably fine for now though.

Reply:
Thanks @jwm4, addressed!